src/nvim/option.c

```diff
@@ -598,23 +598,20 @@ void set_init_1(void)
     }
   }
 
-  /*
-   * 'maxmemtot' and 'maxmem' may have to be adjusted for available memory
-   */
+  // 'maxmemtot' and 'maxmem' are ignored in Neovim, but we set them to the
+  // amount of available memory.
```

Comment:
Now that you mention it, we should remove it entirely if we don't think it would be useful as a read-only value. Because, _setting_ it should be an error: if a plugin or user config is setting the value, they expect the value to have the intended effect, and we don't want to silently ignore their intent.